### PR TITLE
Add AWS Cognito authentication module

### PR DESCRIPTION
**Pull request by ngmihai7**
This PR will add **Cognito** user authentication module which will help us improve the new projects implementations and time estimations.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,57 +19,59 @@
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
     "test:e2e": "jest --config ./test/jest-e2e.json",
-    "prisma:seed": "ts-node prisma/seed.ts"
+    "prisma:migrate": "prisma migrate dev"
```
Comment (GarryOne) [inline package.json]: We need to `seed` command, as well.

File: src/modules/auth/guards/auth.jwt-cognito.guard.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@Injectable()
+export class JwtCognitoAuthGuard extends AuthGuard('jwt-cognito') {
+  constructor(private reflector: Reflector) {
+    super(reflector);
+  }
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
```
Comment (ngmihai7) [inline src/modules/auth/guards/auth.jwt-cognito.guard.ts:21]: I forgot to delete this line here. We can leave just the super call or remove the methods entirely.

File: src/modules/auth/strategies/auth.jwt-cognito.strategy.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  Strategy,
+  'jwt-cognito',
+) {
+  constructor(private authConfig: AuthConfig) {
+    super({
+      secretOrKeyProvider: passportJwtSecret({
+        cache: true,
+        rateLimit: true,
+        jwksRequestsPerMinute: 5,
+        jwksUri: `${authConfig.authority}/.well-known/jwks.json`,
```
Comment (ngmihai7) [inline src/modules/auth/strategies/auth.jwt-cognito.strategy.ts:20]: This was from an example, I don't know it it's good or not, the `jwksUri` part

File: src/modules/user/user.service.ts [hunk 4]
```diff
@@ -23,13 +23,19 @@ export class UserService {
...
   }): Promise<User[]> {
     const { skip, take, cursor, where, orderBy } = params;
-    return this.prisma.user.findMany({
-      skip,
-      take,
-      cursor,
-      where,
-      orderBy,
-    });
+    return [{
```
Comment (ngmihai7) [inline src/modules/user/user.service.ts]: Shoud we leave this commented here ?

**Pull request closed without merging**